### PR TITLE
Fix intermittent connect failure

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixing some errors which are intermittent and sometimes hard to produce

- [serial-backend] callback in `reinitializeConnection` function had wrong condition.
- [firmware-flasher] no need for serial disconnect in `verifyBoard` function if connection is still active. (patience does)
- [ports] fix callback for `writeConfiguration` helper function.

Screenshot from 2023-07-17 23-57-53

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/serial_backend.js
Comment (chmelevskij) [inline src/js/serial_backend.js:764]: lol

**Review by chmelevskij — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]
tested good

**Pull request merged**
